app/shared_channel_service_iface.go

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import "github.com/mattermost/mattermost-server/v5/model"
+
+// SharedChannelServiceIFace is the interface to the shared channel service
+type SharedChannelServiceIFace interface {
```

Comment:
Idiomatic Go suggests this should be defined with the mock, and the mock should accept the concrete type from `Server.GetSharedChannelSyncService`.  e.g.:

`func newMockRemoteClusterService(service ServiceIFace) *mockRemoteClusterService {...}`

I reluctantly did something similar with the logging engine which I've noted to go back and fix.